### PR TITLE
Fixes for building with MinGW

### PR DESCRIPTION
**Pull request by dgarske**
Fixes https://github.com/wolfSSL/wolfMQTT/issues/177.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,6 +41,23 @@ For building wolfMQTT with TLS support in Visual Studio:
...
+
+# wolfSSL
+cd wolfssl
+./configure --host=i686 CC=i686-w64-mingw32-gcc LD=i686-w64-mingw32-ld CFLAGS="-DWIN32 -DMINGW -D_WIN32_WINNT=0x0600" LIBS="-lws2_32" --prefix=$PREFIX --disable-crypttests --disable-examples --disable-shared
+make
+make install
+
+# wolfMQTT
+cd ../wolfmqtt
+./configure --host=i686 CC=i686-w64-mingw32-gcc LD=i686-w64-mingw32-ld CFLAGS="-DWIN32 -DMINGW -D_WIN32_WINNT=0x0600 -DBUILDING_WOLFMQTT" LIBS="-lws2_32" --prefix=$PREFIX --disable-examples --disable-tls --disable-shared
```
Comment (embhorn) [inline README.md:58]: Is `--disable-tls` required for MinGW? It is confusing because the instructions for building wolfSSL are directly above.

**Review by embhorn — COMMENTED** [review]
Looks good, thanks David. Just one question about the configure option.

File: README.md (hunk 1 above)
Comment (dgarske) [reply README.md:58]: Thanks for catching that. I was having trouble linking the cross-compiled static library, which is why its there. I will see if I can resolve that now.

**Review by embhorn — APPROVED** [review]
Excellent, thanks David.

**Pull request merged**